test/common/fuse_crudl.go

```diff
@@ -0,0 +1,94 @@
+package common
+
+import (
+	goctx "context"
+	"fmt"
+	"github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/test/resources"
+	"testing"
+)
+
+var (
+	fuseLoginUser     = fmt.Sprintf("%v-%d", DefaultTestUserName, 0)
+	fuseLoginPassword = DefaultPassword
+)
+
+func loginOpenshift(ctx *TestingContext, masterUrl, username, password, namespacePrefix string) error {
+	authUrl := fmt.Sprintf("%s/auth/login", masterUrl)
+
+	if err := resources.DoAuthOpenshiftUser(authUrl, username, password, ctx.HttpClient, "testing-idp"); err != nil {
+		return err
+	}
+
+	openshiftClient := resources.NewOpenshiftClient(ctx.HttpClient, masterUrl)
+	if err := resources.OpenshiftUserReconcileCheck(openshiftClient, ctx.Client, namespacePrefix, username); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Tests that a user in group rhmi-developers can log into fuse and
+// create an integration
+func TestFuseCrudlPermissions(t *testing.T, ctx *TestingContext) {
+	if err := createTestingIDP(goctx.TODO(), ctx.Client, ctx.HttpClient, ctx.SelfSignedCerts); err != nil {
+		t.Fatalf("error while creating testing idp: %v", err)
+	}
+
+	// get console master url
+	rhmi, err := getRHMI(ctx.Client)
+	if err != nil {
+		t.Fatalf("error getting RHMI CR: %v", err)
+	}
+	masterURL := rhmi.Spec.MasterURL
+
+	// Get the fuse host url from the rhmi status
+	fuseHost := rhmi.Status.Stages[v1alpha1.ProductsStage].Products[v1alpha1.ProductFuse].Host
+
+	err = loginOpenshift(ctx, masterURL, fuseLoginUser, fuseLoginPassword, rhmi.Spec.NamespacePrefix)
+	if err != nil {
+		t.Fatalf("error logging into openshift: %v", err)
+	}
+
+	// Get a client that authenticated to fuse via oauth
+	authenticatedFuseClient, err := resources.ProxyOAuth(ctx.HttpClient, fuseHost, fuseLoginUser, DefaultPassword)
```

Comment:
Do we need to do this full ProxyOAuth flow, if we have already logged the user in via Openshift, if we use the httpClient we can go directly to the syndesis endpoint and select `testing-idp`, we can skip going to `/oauth/start`

Reply:
It looks like I still need to select the testing-idp, fetch the form and submit username and password. Hitting the syndesis url directly instead of oauth/start doesn't seem to work.